pkg/operator/ceph/pool/controller.go

```diff
@@ -127,20 +125,29 @@ func (r *ReconcileCephBlockPool) reconcile(request reconcile.Request) (reconcile
 			return reconcile.Result{}, nil
 		}
 		// Error reading the object - requeue the request.
-		return opcontroller.ImmediateRetryResult, errors.Wrapf(err, "failed to get CephBlockPool")
+		return reconcile.Result{}, errors.Wrapf(err, "failed to get CephBlockPool")
+	}
+
+	// The CR was just created, initializing status fields
+	if cephBlockPool.Status == nil {
+		cephBlockPool.Status = &cephv1.Status{}
+		cephBlockPool.Status.Phase = k8sutil.Created
+		err := opcontroller.UpdateStatus(r.client, cephBlockPool)
```

Comment:
Could the new status be a param to the method? Seems like it would be cleaner for all the places the status is updated. For example:
```
err := opcontroller.UpdateStatus(r.client, cephBlockPool, k8sutil.Created)
```

Reply:
The problem is that if we do this, then `opcontroller.UpdateStatus()` cannot be generic anymore.

Reply:
Is it not just a string? Maybe we should change the type of the phase to a string instead of using a simple type that is anyway a string.

Reply:
We could hardly make it generic. We still need the object to be updated. I'm using `obj runtime.Object` to make `UpdateStatus` generic. Inside the `UpdateStatus` we have no idea of the type of the `obj`. So we can't update it. This has to happen before calling `UpdateStatus`.

Reply:
Ok makes sense thanks